util/oidc/oidc.go

```diff
@@ -175,13 +186,14 @@ func (a *ClientApp) HandleLogin(w http.ResponseWriter, r *http.Request) {
 	}
 	returnURL := r.FormValue("return_url")
 	stateNonce := a.generateAppState(returnURL)
+
 	grantType := InferGrantType(oauth2Config, oidcConf)
 	var url string
 	switch grantType {
 	case GrantTypeAuthorizationCode:
-		url = oauth2Config.AuthCodeURL(stateNonce)
+		url = oauth2Config.AuthCodeURL(stateNonce, opts...)
 	case GrantTypeImplicit:
-		url = ImplicitFlowURL(oauth2Config, stateNonce)
+		url = ImplicitFlowURL(oauth2Config, stateNonce, opts...)
```

Comment:
Even though we pass the `AuthCodeOption` option is passed to `ImplicitFlowURL`, `ImplicitFlowURL` logic is not modifying the URL like AuthCodeURL is doing. See:

```golang
	for _, opt := range opts {
		switch opt {
		case oauth2.AccessTypeOnline:
			v.Set("access_type", "online")
		case oauth2.AccessTypeOffline:
			v.Set("access_type", "offline")
		case oauth2.ApprovalForce:
			v.Set("approval_prompt", "force")
		}
		// I think we need else case here to set the URL
	}
```

Can you fix `ImplicitFlowURL` to also modify the URL so that this will work for CLI logins? To test this, from Argo CLI you can rebuild argocd CLI, run `argocd login --sso` and verify the generated URL has the claims URL parameter.

Reply:
Yes I am trying to solve it. It gets a bit complicated though with the protobuf settings vs the normal settings, reusing code in the util package, unallowed import cycles and my lack of go skills.